werkit/lambda/default_handler.py

```diff
@@ -0,0 +1,11 @@
+import os
+import boto3
+import asyncio
+from .parallel import synchronous_map_on_lambda
+
+def handler(event, context):
```

Comment:
This interface makes sense for testing, however I imagine that in our application the list of things to map over will be either checked into the calling application's code or generated by it. Given that, I think perhaps the application ought to implement the lambda handler, and call these functions itself, rather than write the list to S3 on every request for a separate lambda to pick up.

It would need to pass through the input path (possibly as a signed URL?) from its own invocation, and provide its own list of keys for the mapping. To facilitate this, perhaps `synchronous_map_on_lambda` should accept an additional parameter with arguments that are forwarded to each child Lambda invocation.

Reply:
>> This interface makes sense for testing, however I imagine that in our application the list of things to map over will be either checked into the calling application's code or generated by it. Given that, I think perhaps the application ought to implement the lambda handler, and call these functions itself, rather than write the list to S3 on every request for a separate lambda to pick up.

I wanted to set this up so that `default_handler.handler` and `s3_handler.handler` could be run directly on lambda, without outside code needed to set it up.                                                                       

But the calling application could also import the `default_handler.handler` function, and call it directly. For example, this is how `s3_handler.handler` is implemented (it imports and calls `default_handler.handler`). This way, you would not need to write the args for `default_handler.handler` to S3.

Reply:
I imagine the way this will work in our application is that we will receive a file posted to S3. The list of keys will be checked into our application code. The application will prepare a list of args to pass to function `parallel_map_on_lambda`. For our application, this will basically be:

`lambda url_to_file: [(key, (key, url_to_file)) for key in keys]`

Note that this data structure (a list of pairs, with a string as a first element), is sufficient to encode a dict, and can be passed to the python dict and OrderedDict constructors.  

The first item in the pair (`key`) is the name of the key that would be set in the final returned dict. The second value in the pair (` (key, url_to_file)`) are the args that will be passed into the sub-lambda. 

The above lambda is application-specific, and would get defined in our application code. I imagine we would add an optional parameter to `s3_handler.handler`, that allows application-specific code to intercept the value uploaded to s3 and transform it before passing it to `default_handler.handler`, and then into `parallel_map_on_lambda`.